SA3/qa_web/urls.py

```diff
@@ -13,6 +13,7 @@
     path('vote/', views.vote, name='vote'),
     path('questions/<int:id_>/edit/', views.edit, name='edit'),
     url(r'QuestionIndex/', views.QuestionDisplayView.as_view(), name='QuestionIndex'),
+    url(r'^tag/(?P<tag>[-\w]+)/$', views.QuestionsByTagView.as_view(), name='question_by_tag'),
```

Comment:
`(?P<tag>[-\w]+)` This regex call is blowing my mind, I still don't understand it. But great job, as it works!

Reply:
Bro you can see the comment I add on views.py, around line 396. The code is:
return Questions.objects.order_by('-creation_date').filter(tag__slug=self.kwargs['tag'])\
You see I'm using self.kwargs['tag'] which I get through using regex. It's very efficient to use regex on url to catch parameters.
I gave the comment above that call:
# that's why we should use regex group name in url: url(r'^tag/(?P<tag>[-\w]+)/$'. A common way to get parameters from front end.

Reply:
Very nice! I can see it in the you preparing the event in line 78 of `question_display_page.html`. This is very interesting.